Fix C style struct initialization that is invalid C++

For some reason, g++ 5.4 was accepting this, although tag-based struct
initialization is AFAIK still invalid even in C++17.
g++ 4.6 is less permissive, and refuses to compile this.

diff --git codels/pom_io_codels.cc codels/pom_io_codels.cc
index 7acd00c..67f56db 100644
--- codels/pom_io_codels.cc
+++ codels/pom_io_codels.cc
@@ -57,9 +57,9 @@ pom_io_start(pom_ids *ids, const genom_context self)
   ids->max_jerk = ukf::state_s::max_dadt;
   ids->max_dw = ukf::state_s::max_dwdt;

-  ids->offset = (pom_ids_offset_s){
-    .p = { 0., 0., 0.}, .q = { 0., 0., 0., 1. }
-  };
+  ids->offset.p[0] = ids->offset.p[1] = ids->offset.p[2] = 0.;
+  ids->offset.q[0] = ids->offset.q[1] = ids->offset.q[2] = 0.;
+  ids->offset.q[3] = 1.;

   /* init logging */
   ids->log = new pom_log_s;
